cmake/dependencies.cmake

```diff
@@ -108,6 +108,18 @@ FetchContent_Declare(
   SYSTEM
   SOURCE_SUBDIR crates/c-api)
 
+FetchContent_GetProperties(ada)
+if(NOT ada_POPULATED)
+  # Disable git to prevent CPM from downloading 3rdparty repositories.
+  # An intentional side effect is that tests and tools are not built
+
+  FetchContent_Populate(ada)
+  set(GIT_EXECUTABLE_OLD ${GIT_EXECUTABLE})
+  set(GIT_EXECUTABLE "" CACHE INTERNAL "Invalid path to the Git executable")
+  add_subdirectory(${ada_SOURCE_DIR} ${ada_BINARY_DIR})
+  set(GIT_EXECUTABLE ${GIT_EXECUTABLE_OLD} CACHE STRING "location of git" FORCE)
```

Comment:
This seems not fun.

Should we be setting `BUILD_TESTING` to `OFF` instead of this hack around disabling the `git` exec lookup?

Additionally we should be able to `set(ADA_TOOLS OFF)` to disable building the tools right? That feels like a slightly better solution, but ideally they had an option like `ADA_TESTING` instead of `BUILD_TESTING` right?

I guess this ends up being net less because we'll have to do the same `BUILD_TESTING` hack to disable and re-enable as we only want tests in our repo.

Reply:
I was seeing some git activity, and was trying to get rid of: "detected dubious ownership in repository at"

Reply:
I'm ambivalent. Ideally we just did something like:

`set(ADA_TOOLS OFF)` and `set(ADA_TESTING OFF)` and called it a day (without the set it back hack). I'm fine with this, but would love Noah's opinion here. I'm a CMake newbie.

Reply:
> I was seeing some git activity, and was trying to get rid of: "detected dubious ownership in repository at"

Did this fix it?

Reply:
There's no git activity now! But same error.

Reply:
Bummer...

It looks like they distribute a single header version in releases... Is it easy in CMake to download that and add it as a target? 

https://github.com/ada-url/ada/releases/tag/v2.7.2

Anyways I'm not trying to add more work for you, we can just start with this.

Reply:
@dotnwat  Any objection to merging this as-is to mitigate [this type of failure](https://buildkite.com/redpanda/redpanda/builds/40273#018b8dea-65c0-4a31-8c45-e0b3d41a1c30https://buildkite.com/redpanda/redpanda/builds/40273#018b8dea-65c0-4a31-8c45-e0b3d41a1c30) and then reworking the build in followup work?